# İP6 — Tutarsızlık Analizi ve Karşılaştırmalı Validasyon

**Görev (Bölüm 10, Görev 8):** Tutarsızlık Analizi ve Karşılaştırmalı Validasyon
**Sorumlu:** Samet Koca | **Tarih:** 18 Temmuz 2026

**Girdi:** `data/processed/master_akts_final.xlsx` (İP5 çıktısı) + `data/processed/master_bloom.xlsx`

Bu notebook, metodolojideki Aşama 2 (Tutarsızlık Analizi) ve Aşama 4 (Karşılaştırma/Validasyon) adımlarını ve H1/H2/H4 hipotez testlerini çalıştırır. **H3, öğrenci iş yükü anketi (Kanal C) toplanmadığı için test edilememektedir** — bkz. rapor Sınırlılıklar bölümü.

In [1]:
import sys
from pathlib import Path

BASE = Path.cwd().resolve().parents[1]
sys.path.insert(0, str(BASE / "ip6-validasyon" / "src"))

import ip6_validasyon as ip6

ip6.main()

İP6 — TUTARSIZLIK ANALİZİ VE KARŞILAŞTIRMALI VALİDASYON



Yüklendi: 4133 ders (master_akts_final.xlsx + master_bloom.xlsx birleşimi)

------------------------------------------------------------
AŞAMA 2(a) / H2 — Kurum-İçi Tutarsızlık (aynı isimli ders, farklı bölüm)
------------------------------------------------------------
  Aynı isimle birden fazla bölümde geçen ders sayısı: 244
  Bunlardan ±2 AKTS aralığını AŞANLAR: 28 (11.5%)
  Ortalama AKTS aralığı (bu gruplarda): 1.21

------------------------------------------------------------
AŞAMA 2(b) — Etkinlik Bazlı Etki
------------------------------------------------------------
  has_lab        var=+15.97%  yok=+9.01%  fark=+6.96%  p=0.0005
  has_proje      var=+15.54%  yok=+8.75%  fark=+6.79%  p=0.0000
  has_uygulama   var=+12.37%  yok=+8.93%  fark=+3.44%  p=0.0001
  has_atolye     var=+17.89%  yok=+9.20%  fark=+8.69%  p=0.3239
  has_odev       var=+11.28%  yok=+8.82%  fark=+2.46%  p=0.0419
  has_sinav      var=+9.39%  yok=+3.11%  fark=+6.28%  p=0.0028
  has_sunum      var=+13.59%  yok=+8

### Çıktı Dosyalarını İncele

In [2]:
import pandas as pd

pd.read_excel(BASE / 'ip6-validasyon' / 'ayni_isimli_ders_tutarsizligi.xlsx').head(10)

,DersAdi,n_bolum,n_kayit,akts_min,akts_max,akts_std,akts_araligi
0,Bitirme Çalışması,17,17,1,16,3.314407,15
1,Spor Bilimlerinde Araştırma Projesi I,2,2,4,10,4.242641,6
2,Bilgisayar Grafikleri,2,2,2,6,2.828427,4
3,Bitirme Projesi,3,3,2,6,2.081666,4
4,Osmanlı Türkçesi I,2,2,2,6,2.828427,4
5,Osmanlı Türkçesi II,2,2,2,6,2.828427,4
6,Mesleki İngilizce I,7,7,2,6,1.676163,4
7,Mesleki İngilizce II,7,7,2,6,1.345185,4
8,Yazarlık I,2,2,3,7,2.828427,4
9,Mesleki İngilizce III,2,2,2,6,2.828427,4


In [3]:
pd.read_excel(BASE / 'ip6-validasyon' / 'etkinlik_bazli_etki.xlsx')

,etkinlik,n_var,ort_sapma_pct_var,n_yok,ort_sapma_pct_yok,fark,mannwhitney_p,anlamli_mi(p<0.05)
0,has_atolye,20,17.89,4113,9.20,8.69,0.323891,False
1,has_lab,139,15.97,3994,9.01,6.96,0.000535,True
2,has_proje,303,15.54,3830,8.75,6.79,0.000000,True
3,has_sinav,4039,9.39,94,3.11,6.28,0.002847,True
4,has_sunum,312,13.59,3821,8.89,4.70,0.081982,False
5,has_uygulama,382,12.37,3751,8.93,3.44,0.000054,True
6,has_odev,723,11.28,3410,8.82,2.46,0.041852,True


In [4]:
pd.read_excel(BASE / 'ip6-validasyon' / 'inceleme_kuyrugu_ilk50.xlsx').head(10)

,Katalogid,DersAdi,Fakulte,AKTSKredi,adaptif_akts,sapma_yuzde,gerekce
0,1701232,Araştırma Projesi,Spor Bilimleri,3,11.2826,276.09,AKTS_mevcut=3.0 | model_tahmin=14.29 | kural_a...
1,207536,Sayısal Tasarım Laboratuvarı,Mühendislik,1,3.6051,260.51,AKTS_mevcut=1.0 | model_tahmin=4.54 | kural_ak...
2,103381,Bitirme Çalışması,Fen-Edebiyat,1,2.8331,183.31,AKTS_mevcut=1.0 | model_tahmin=3.49 | kural_ak...
3,711070,Bitirme Çalışması,Güzel Sanatlar,2,4.8777,143.88,AKTS_mevcut=2.0 | model_tahmin=5.75 | kural_ak...
4,731005,Teknik Çizim Perspektif I,Güzel Sanatlar,1,2.4238,142.38,AKTS_mevcut=1.0 | model_tahmin=3.04 | kural_ak...
5,9509115,Üniversite ve Yaşam,Eğitim,3,7.1360,137.87,AKTS_mevcut=3.0 | model_tahmin=8.86 | kural_ak...
6,113108,Mesleki İngilizce I,Fen-Edebiyat,2,4.6934,134.67,AKTS_mevcut=2.0 | model_tahmin=5.67 | kural_ak...
7,721117,Çağdaş Grafik Tasarım Tarihi,Güzel Sanatlar,2,4.6803,134.01,AKTS_mevcut=2.0 | model_tahmin=5.73 | kural_ak...
8,713167,Desen ve Anatomi I,Güzel Sanatlar,2,4.2967,114.84,AKTS_mevcut=2.0 | model_tahmin=5.18 | kural_ak...
9,9507021,Doğa Fotoğrafı,Güzel Sanatlar,2,4.2231,111.16,AKTS_mevcut=2.0 | model_tahmin=5.10 | kural_ak...


### Bulgular

- **H1 DESTEKLENİYOR:** Lab/proje içeren derslerin ortalama sapması (+19.9%), diğer derslerden (+8.2%) anlamlı derecede yüksek (Mann-Whitney p<0.001).
- **H4 DESTEKLENİYOR:** Bloom düzeyi ile sapma arasında pozitif, anlamlı bir ilişki var (Spearman rho=0.16, p<0.001; Kruskal-Wallis p<0.001).
- **H2 (proxy):** Kurum-içi aynı isimli derslerin %11.5'i ±2 AKTS aralığını aşıyor — kurumlar arası (Kanal B) veri toplanınca yeniden test edilmeli.
- **H3: test edilemedi** (ground truth/anket verisi yok).

Tam rapor: `ip6-validasyon/IP6_SONUC_RAPORU.md`.